component/component.go

```diff
@@ -67,18 +68,25 @@ type Factory interface {
 // ConfigUnmarshaler interface is an optional interface that if implemented by a Factory,
 // the configuration loading system will use to unmarshal the config.
 type ConfigUnmarshaler interface {
-	// Unmarshal is a function that un-marshals a viper data into a config struct in a custom way.
-	// componentViperSection *viper.Viper
-	//   The config for this specific component. May be nil or empty if no config available.
+	// Unmarshal is a function that loads data into a config struct in a custom way.
+	// componentSection *configload.Parser
+	//   The parser for this specific component. May be nil or empty if no config available.
 	// intoCfg interface{}
 	//   An empty interface wrapping a pointer to the config struct to unmarshal into.
-	Unmarshal(componentViperSection *viper.Viper, intoCfg interface{}) error
+	Unmarshal(componentSection *configload.Parser, intoCfg interface{}) error
 }
 
-// CustomUnmarshaler is a function that un-marshals a viper data into a config struct
+// CustomUnmarshaler is a function that loads data into a config struct
 // in a custom way.
-// componentViperSection *viper.Viper
-//   The config for this specific component. May be nil or empty if no config available.
+// componentSection *configload.Parser
+//   The parser for this specific component. May be nil or empty if no config available.
 // intoCfg interface{}
 //   An empty interface wrapping a pointer to the config struct to unmarshal into.
-type CustomUnmarshaler func(componentViperSection *viper.Viper, intoCfg interface{}) error
+type CustomUnmarshaler func(componentSection *configload.Parser, intoCfg interface{}) error
+
+// ToCustomUnmarshaler creates a custom unmarshaler from a Viper unmarshaler.
+func ToCustomUnmarshaler(viperUnmarshaler func(*viper.Viper, interface{}) error) CustomUnmarshaler {
```

Comment:
Can we split for the moment this PR into 2:
1. All changes except changes in component.go (we keep the old interface for the moment), but we add the Parser and move helpers for viper.
2. Changes in the component.go (reason is maybe we just don't do these changes anymore and jump to add the new unmarshaler directly on the config, and deal that's it).